### PR TITLE
WIP: Implement a minimal `interruptible` feature

### PR DESCRIPTION
**Pull request by guizmaii**
Hi @skjolber @tyrcho,

Use case: We're launching in parallel some calls to `pack` and we want to stop them when one of them succeed.

You already have a mechanism of interruption thanks to your `deadline` feature, so I transplanted my mecanism with yours.

I still have to write some tests.

### CODE REVIEW

File: src/main/java/com/github/skjolberg/packing/Constants.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.github.skjolberg.packing;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public interface Constants {
+
+	AtomicBoolean ALWAYS_FALSE = new AtomicBoolean(false);
```
Comment (skjolber) [inline src/main/java/com/github/skjolberg/packing/Constants.java]: Could this constant perhaps go into the Packager instead - so no Constants interface?

**Comment by skjolber** [comment]
This is a good feature, I have been thinking about multi-threading, but not really had a good use for it in my own applications. Although the deadline and interrupt can be combined in a single interface, adding a parameter for the interrupt (atomicboolean) is really just as good.

**Comment by guizmaii** [comment]
> Although the deadline and interrupt can be combined in a single interface, adding a parameter for the interrupt (atomicboolean) is really just as good.

Yes, a better, more general solution could be to pass a function: 
```scala
def pack(..., interrupt: () => Boolean)
```

(Scala notation, sorry. I don't know how to write that with Java). 

I can implement that if you want.

File: src/main/java/com/github/skjolberg/packing/Constants.java (hunk 1 above)
Comment (guizmaii) [reply src/main/java/com/github/skjolberg/packing/Constants.java]: Yes, why not.

**Pull request closed without merging**